docs/userguides/config.md

```diff
@@ -45,6 +45,19 @@ dependencies:
 
 will download the [Open Zeppelin](https://github.com/OpenZeppelin/openzeppelin-contracts) package with version `4.4.2`.
 
+**WARN**: The `version:` key only works when the version is published to GitHub; use the `branch:` key when the version is only available by tag.
+
+For example, to install a version based on a `git` tag, do the following:
+
+```yaml
+dependencies:
+  - name: Uniswap
+    github: Uniswap/v3-core
+    branch: v1.0.0
+```
+
+The `branch:` config installs the code from that branch or tag whereas the `version:` config uses the official GitHub release API.
```

Comment:
Wonder if `branch` is the confusing bit and it should be `ref` or something (breaking change obvs)

Reply:
We could deprecate `branch:`, add `ref:`, make `branch:` be an alias and undocument it, and then remove `branch:` on 0.7. We could do that now.